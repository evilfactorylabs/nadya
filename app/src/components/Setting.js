import React, { useState, useEffect } from 'react'
import dayjs from 'dayjs'
import CloseIcon from '@material-ui/icons/Close'

import {
  Snackbar,
  IconButton,
  Button,
  Card,
  CardActions,
  CardContent,
  TextField,
  Avatar,
  Grid
} from '@material-ui/core'

import { deepOrange } from '@material-ui/core/colors'
import { makeStyles } from '@material-ui/core/styles'

const useStyles = makeStyles(theme => ({
  avatar: {
    width: 90,
    height: 90,
    fontSize: 48,
    marginTop: theme.spacing(4),
    backgroundColor: deepOrange[500],
    color: '#fff'
  },
  input: {
    marginTop: theme.spacing(2)
  },
  action: {
    marginLeft: 'auto',
    marginRight: theme.spacing(1),
    marginBottom: theme.spacing(1)
  }
}))

export default ({ user, updateUser }) => {
  const classes = useStyles()
  const [snackbarOpen, setSnackbarOpen] = useState(false)
  const [userName, setUserName] = useState('')

  const handleSave = () => {
    updateUser(userName)
      .then(() => {
        setSnackbarOpen(true)
      })
      .catch(err => {
        throw err
      })
  }

  useEffect(() => {
    setUserName(user.name)
  }, [user])

  return (
    <>
      <Card>
        <Grid container justify='center' alignItems='center'>
          <Avatar className={classes.avatar}>
            {userName ? userName.charAt(0).toUpperCase() : 'N'}
          </Avatar>
        </Grid>
        <CardContent>
          <TextField
            className={classes.input}
            variant='outlined'
            value={user._id}
            label='User ID'
            margin='dense'
            fullWidth={true}
            disabled={true}
          />
          <TextField
            className={classes.input}
            variant='outlined'
            value={dayjs(user.created_at).format('DD MMMM YYYY')}
            label='Member sejak'
            margin='dense'
            fullWidth={true}
            disabled={true}
          />
          <TextField
            className={classes.input}
            variant='outlined'
            value={userName || ''}
            label='Nama'
            margin='dense'
            onChange={e => setUserName(e.target.value)}
            fullWidth={true}
          />
        </CardContent>
        <CardActions>
          <Button
            className={classes.action}
            color='primary'
            variant='contained'
            onClick={handleSave}
          >
            Simpan
          </Button>
        </CardActions>
      </Card>
      <Snackbar
        anchorOrigin={{
          vertical: 'bottom',
          horizontal: 'right'
        }}
        open={snackbarOpen}
        autoHideDuration={3000}
        onClose={() => setSnackbarOpen(false)}
        message={<span id='message-id'>Data berhasil disimpan</span>}
        action={[
          <IconButton
            key='close'
            aria-label='Close'
            color='inherit'
            className={classes.close}
            onClick={() => setSnackbarOpen(false)}
          >
            <CloseIcon />
          </IconButton>
        ]}
      />
    </>
  )
}
