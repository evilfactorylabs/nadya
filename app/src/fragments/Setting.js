import React, { Component } from 'react'
import Setting from '../components/Setting'

import { updateUser, getUser, getUserIdFromLS } from '../services/user'

class SettingFragment extends Component {
  state = {
    userId: '',
    userName: '',
    user: {}
  }

  _getUser = userId => {
    getUser(userId)
      .then(user => {
        this.setState({
          user,
          userId,
          userName: user.name
        })
      })
      .catch(err => {
        throw err
      })
  }

  _updateUser = userName => {
    const { userId, user } = this.state
    const payload = {
      _id: userId,
      _rev: user._rev,
      createdAt: user.createdAt,
      name: userName
    }
    return new Promise((resolve, _) => {
      updateUser(payload)
        .then(_ => {
          this._getUser(userId)
          resolve()
        })
        .catch(err => {
          throw err
        })
    })
  }

  componentDidMount() {
    const userId = getUserIdFromLS()
    this._getUser(userId)
  }

  render() {
    return <Setting updateUser={this._updateUser} user={this.state.user} />
  }
}

export default SettingFragment
