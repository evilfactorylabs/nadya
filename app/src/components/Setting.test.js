import Setting from './Setting'

afterAll(async () => {
  await cleanup()
})

describe('Fragments: Setting.js', () => {
  test('It should render Setting screen', () => {
    const user = {
      _id: 'xxx'
    }
    const updateUser = () => {}
    const { container } = render(
      <Setting user={user} updateUser={updateUser} />
    )

    expect(container).toBeInTheDocument()
    expect(container).toMatchSnapshot()
  })
})
