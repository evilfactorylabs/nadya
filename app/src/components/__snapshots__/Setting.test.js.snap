// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fragments: Setting.js It should render Setting screen 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation1 MuiCard-root MuiPaper-rounded"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
    >
      <div
        class="MuiAvatar-root makeStyles-avatar-1 MuiAvatar-colorDefault"
      >
        N
      </div>
    </div>
    <div
      class="MuiCardContent-root"
    >
      <div
        class="MuiFormControl-root MuiTextField-root makeStyles-input-2 MuiFormControl-marginDense MuiFormControl-fullWidth"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-marginDense MuiInputLabel-outlined Mui-disabled Mui-disabled MuiFormLabel-filled"
          data-shrink="true"
        >
          User ID
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root Mui-disabled Mui-disabled MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
        >
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-193 MuiOutlinedInput-notchedOutline"
            style="padding-left: 8px;"
          >
            <legend
              class="PrivateNotchedOutline-legend-194"
              style="width: 0px;"
            >
              <span>
                ​
              </span>
            </legend>
          </fieldset>
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input Mui-disabled Mui-disabled MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
            disabled=""
            type="text"
            value="xxx"
          />
        </div>
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root makeStyles-input-2 MuiFormControl-marginDense MuiFormControl-fullWidth"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-marginDense MuiInputLabel-outlined Mui-disabled Mui-disabled MuiFormLabel-filled"
          data-shrink="true"
        >
          Member sejak
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root Mui-disabled Mui-disabled MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
        >
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-193 MuiOutlinedInput-notchedOutline"
            style="padding-left: 8px;"
          >
            <legend
              class="PrivateNotchedOutline-legend-194"
              style="width: 0px;"
            >
              <span>
                ​
              </span>
            </legend>
          </fieldset>
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input Mui-disabled Mui-disabled MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
            disabled=""
            type="text"
            value="02 July 2019"
          />
        </div>
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root makeStyles-input-2 MuiFormControl-marginDense MuiFormControl-fullWidth"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense MuiInputLabel-outlined"
          data-shrink="false"
        >
          Nama
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
        >
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-193 MuiOutlinedInput-notchedOutline"
            style="padding-left: 8px;"
          >
            <legend
              class="PrivateNotchedOutline-legend-194"
              style="width: 0.01px;"
            >
              <span>
                ​
              </span>
            </legend>
          </fieldset>
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
    <div
      class="MuiCardActions-root MuiCardActions-spacing"
    >
      <button
        class="MuiButtonBase-root MuiButton-root makeStyles-action-3 MuiButton-contained MuiButton-containedPrimary"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Simpan
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;
